frozen_solid/aws-opensearchserverless-is-public.yaml

```
@@ -0,0 +1,54 @@
+rules:
+- id: aws-opensearchserverless-is-public
+  patterns:
+  - pattern-inside: |
+      resource "aws_opensearchserverless_security_policy" $ANYTHING {
+        ...
+        type =  "network"
+        ...
+      }
+  - pattern-either:
+    - patterns:
+      - pattern: policy = $JSONPOLICY
+      - metavariable-pattern:
+          metavariable: $JSONPOLICY
+          language: json
+          pattern: |
+            {..., "AllowFromPublic":true, ... }
+    - patterns:
+      - pattern-inside: policy = jsonencode(...)
+      - pattern: |
+          {..., AllowFromPublic = true, ...}
+    - patterns:
+      - pattern-inside: policy = jsonencode([...])
+      - pattern: |
+          {..., AllowFromPublic = true, ...}
+    - patterns:
+      - pattern-inside: policy = jsonencode([...])
+      - pattern: "{..., \"AllowFromPublic\" : true, ...}                 \n"
+  message: The opensearch serverless collection is accessible from the public internet
+    using a public VPC endpoint.  To fix this, set your `AllowFromPublic` to `"false"`
+    and specify an internal VPC endpoint in `SourceVPCEs` .
+  metadata:
+    category: security
```

```suggestion
    category: security
    subcategory:
    - vuln
```

Since you have selected the category security, you have to also include a subcategory. 
See the docs: https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/#subcategory

---

Oh nevermind. I did not see you had it further down.